presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -42,18 +43,23 @@
 public class IterativeOptimizer
         implements PlanOptimizer
 {
-    private final List<PlanOptimizer> legacyRules;
+    private final Optional<List<PlanOptimizer>> legacyRules;
     private final RuleIndex ruleIndex;
     private final StatsRecorder stats;
 
     public IterativeOptimizer(StatsRecorder stats, Set<Rule<?>> rules)
     {
-        this(stats, ImmutableList.of(), rules);
+        this(stats, Optional.empty(), rules);
     }
 
     public IterativeOptimizer(StatsRecorder stats, List<PlanOptimizer> legacyRules, Set<Rule<?>> newRules)
```

Comment:
This commit changes how `new IterativeOptimizer(stats, emptyList(), someRules)` behaves:
- previously it would assume there are no legacy optimizers and still apply rules
- now it would assume there where fallback legacy rules (just an empty list)

However, no such instance is constructed in our code, so I think this is not a problem.

Reply:
Another thing is that named factory methods would be more explicit than overloaded constructor. I can do this if we want this change merged.